assets/app/scripts/controllers/services.js

```diff
@@ -24,7 +25,21 @@ angular.module('openshiftConsole')
       $scope.emptyMessage = "No services to show";
       updateFilterWarning();
       console.log("services (subscribe)", $scope.unfilteredServices);
-    }));    
+    }));
+
+    watches.push(DataService.watch("routes", $scope, function(routes){
+        $scope.routesByService = routesByService(routes.by("metadata.name"));
+        console.log("routes (subscribe)", $scope.routesByService);
+    }));
+
+    var routesByService = function(routes) {
+        var routeMap = {};
+        angular.forEach(routes, function(route, routeName){
+            routeMap[route.serviceName] = routeMap[route.serviceName] || {};
+            routeMap[route.serviceName][routeName] = route
```

Comment:
damn golang, it has left me with sloppy semi colons after all my mental retraining to not put them on every line.  Fix incoming